### DISCUSSION TITLE
FR: persistence of emitters

### DISCUSSION THREAD

**Discussion by thiswillbeyourgithub** [open-webui/open-webui#5026] [Ideas]
**Is your feature request related to a problem? Please describe.**
When debugging or inspecting what's going on in a filter or function, it's pretty hard to do because each emitter removes the previous emitter.

**Describe the solution you'd like**
I think it would be better if they stacked up on top of each but only the last one would be visible, the other ones would appear after clicking an arrow.
A bit like this:
<details><summary>last emitter</summary>

- first emitter

- second emitter

- last emitter
</details>

**Describe alternatives you've considered**
Digging through the logs

**Comment by PlebeiusGaragicus** [comment]
I agree that we should persist emitters!  But, I feel differently about them stacking up.

Either way, I wonder how these would persist in message history.

In my use case an agent would emit status as it processes.  I'd like these to be included in the message history somehow.  Think of them as O1's reasoning tokens.
